src/compiler/checker.ts

```diff
@@ -6656,6 +6727,13 @@ module ts {
             if (contextualMapper === identityMapper && isContextSensitive(node)) {
                 return anyFunctionType;
             }
+            var isAsync = !!(node.flags & NodeFlags.Async);

```

Comment:
Consider making a function in `utilities.ts`, it'll hopefully get inlined anyway.